File: cpp/unittests/KthLargestElementInAnArrayTest.cpp
#include "KthLargestElementInAnArray.h"
#include "gtest/gtest.h"

using namespace lcpp;

TEST(KthLargestElementInAnArray, Solution215_1) {
  std::vector<int> Nums1{3, 2, 1, 5, 6, 4};
  const int K1 = 2, Expected1 = 5;
  EXPECT_EQ(Expected1, Solution215_1().findKthLargest(Nums1, K1));

  std::vector<int> Nums2{3, 2, 3, 1, 2, 4, 5, 5, 6};
  const int K2 = 4, Expected2 = 4;
  EXPECT_EQ(Expected2, Solution215_1().findKthLargest(Nums2, K2));
}

File: java/src/test/java/io/binac/leetcode/KthLargestElementInAnArrayTest.java
package io.binac.leetcode;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class KthLargestElementInAnArrayTest {
    private static final KthLargestElementInAnArray.Solution1 SOLUTION1 = new KthLargestElementInAnArray.Solution1();
    private static final KthLargestElementInAnArray.Solution2 SOLUTION2 = new KthLargestElementInAnArray.Solution2();

    private final int[] NUMS1 = {3, 2, 1, 5, 6, 4};
    private final int K1 = 2;
    private final int EXPECTED1 = 5;

    private final int[] NUMS2 = {3, 2, 3, 1, 2, 4, 5, 5, 6};
    private final int K2 = 4;
    private final int EXPECTED2 = 4;

    @Test
    void test1() {
        assertEquals(EXPECTED1, SOLUTION1.findKthLargest(NUMS1, K1));
        assertEquals(EXPECTED2, SOLUTION1.findKthLargest(NUMS2, K2));
    }

    @Test
    void test2() {
        assertEquals(EXPECTED1, SOLUTION2.findKthLargest(NUMS1, K1));
        assertEquals(EXPECTED2, SOLUTION2.findKthLargest(NUMS2, K2));
    }
}